pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/OpChainStats.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.query.runtime.operator;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Suppliers;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Supplier;
+import org.apache.pinot.spi.accounting.ThreadResourceUsageProvider;
+
+
+/**
+ * {@code OpChainStats} tracks execution statistics for {@link OpChain}s.
+ */
+public class OpChainStats {
```

Comment:
i am not familiar with this and how the ThreadResourceUsageProvider is being used. maybe @siddharthteotia @jasperjiaguo can comment on this. thanks

Reply:
I think it is fine to use this provider here. 

Curious where exactly are you going to call `getThreadTimeNs` (accounted cpu time) and `getThreadAllocatedBytes` (accounted memory) because that is what it is being used for by the accountant in the current engine

Reply:
for now it's only in the `toString` method, but later we can use it when reporting metrics

Reply:
So, if we plan on using the accountant and resource usage (cpu time and memory) provider in the multi stage engine, I don't think this wiring is enough. 

Based on some recent discussion, I am guessing there is plan / desire to integrate it into multi stage engine ?

We will have to setup anchor points and collect stats from all the worker threads executing for intermediary layer and above. Leaf layer is already taken care of.

Reply:
> for now it's only in the toString method, but later we can use it when reporting metrics

Got it. My comment crossed with yours. I think we are on the same page that full integration of accounting has to be done separately

Reply:
cc @walterddr about plans to integrate with the multi-stage engine, the reason it's in this PR is because it was moved around during refactoring (first in #9615 and then in this PR) - I'm happy to change it to a simple stopwatch to get time executing on the thread, as with the scheduler design it's pretty trivial to get that information